journalbeat-HEAD crashes with interface conversion: interface is string, not common.Time

I noticed that the latest journalbeat release (5.1.2) doesn't work with the latest elasticsearch (5.4). So I upgraded journalbeat to the latest HEAD (7490ae94018b97fe73196c0417a630a4f3393df4). Unfortunately journalbeat now crashes with the following message:

```
May 24 17:03:18 hera systemd[1]: Starting Journalbeat log shipper...
May 24 17:03:18 hera systemd[1]: Started Journalbeat log shipper.
May 24 17:03:18 hera journalbeat[2202]: panic: interface conversion: interface is string, not common.Time
May 24 17:03:18 hera journalbeat[2202]: goroutine 31 [running]:
May 24 17:03:18 hera journalbeat[2202]: panic(0x8ee800, 0xc4208a2140)
May 24 17:03:18 hera journalbeat[2202]:         /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-go-1.7.4/share/go/src/runtime/panic.go:500 +0x1a1
May 24 17:03:18 hera journalbeat[2202]: github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch.getIndex(0xc42028a1e0, 0xe1efa0, 0xc420169680, 0x0, 0x0)
May 24 17:03:18 hera journalbeat[2202]:         /tmp/nix-build-journalbeat-5.1.2.drv-0/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch/client.go:367 +0xb7
May 24 17:03:18 hera journalbeat[2202]: github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch.createEventBulkMeta(0xe1efa0, 0xc420169680, 0x0, 0xc42028a1e0, 0x0, 0x0, 0x0)
May 24 17:03:18 hera journalbeat[2202]:         /tmp/nix-build-journalbeat-5.1.2.drv-0/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch/client.go:322 +0xbd
May 24 17:03:18 hera journalbeat[2202]: github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch.bulkEncodePublishRequest(0x7f5280006000, 0xc42022d240, 0xe1efa0, 0xc420169680, 0x0, 0xc420174380, 0x32, 0x32, 0x8bc5c0, 0xc420278db0, ...)
May 24 17:03:18 hera journalbeat[2202]:         /tmp/nix-build-journalbeat-5.1.2.drv-0/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch/client.go:289 +0xc5
May 24 17:03:18 hera journalbeat[2202]: github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch.(*Client).PublishEvents(0xc4201843c0, 0xc420174380, 0x32, 0x32, 0x0, 0x1, 0x0, 0x0, 0x42f3c3)
May 24 17:03:18 hera journalbeat[2202]:         /tmp/nix-build-journalbeat-5.1.2.drv-0/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch/client.go:241 +0x13b
May 24 17:03:18 hera journalbeat[2202]: github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single.(*Mode).PublishEvents.func1(0xc4202349c0)
May 24 17:03:18 hera journalbeat[2202]:         /tmp/nix-build-journalbeat-5.1.2.drv-0/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single/single.go:89 +0x7a
May 24 17:03:18 hera journalbeat[2202]: github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single.(*Mode).publish(0xc4202349c0, 0xe25020, 0xc420356000, 0xc4200e1401, 0xc420853ad8, 0x0, 0x0)
May 24 17:03:18 hera journalbeat[2202]:         /tmp/nix-build-journalbeat-5.1.2.drv-0/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single/single.go:144 +0x2aa
May 24 17:03:18 hera journalbeat[2202]: github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single.(*Mode).PublishEvents(0xc4202349c0, 0xe25020, 0xc420356000, 0x1, 0xc420174380, 0x32, 0x32, 0x1, 0x1)
May 24 17:03:18 hera journalbeat[2202]:         /tmp/nix-build-journalbeat-5.1.2.drv-0/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single/single.go:99 +0x74
May 24 17:03:18 hera journalbeat[2202]: github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch.(*elasticsearchOutput).BulkPublish(0xc42016ac00, 0xe25020, 0xc420356000, 0x1, 0xc420174380, 0x32, 0x32, 0xc42003ff88, 0xc420152820)
May 24 17:03:18 hera journalbeat[2202]:         /tmp/nix-build-journalbeat-5.1.2.drv-0/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch/output.go:415 +0x79
May 24 17:03:18 hera journalbeat[2202]: github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher.(*outputWorker).sendBulk(0xc42015e960, 0xc420853c88, 0xc420174380, 0x32, 0x32)
May 24 17:03:18 hera journalbeat[2202]:         /tmp/nix-build-journalbeat-5.1.2.drv-0/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher/output.go:112 +0x125
May 24 17:03:18 hera journalbeat[2202]: github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher.(*outputWorker).onBulk(0xc42015e960, 0xc420853c88, 0xc420174380, 0x32, 0x32)
May 24 17:03:18 hera journalbeat[2202]:         /tmp/nix-build-journalbeat-5.1.2.drv-0/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher/output.go:88 +0x259
May 24 17:03:18 hera journalbeat[2202]: github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher.(*outputWorker).onMessage(0xc42015e960, 0x0, 0x1, 0xe25020, 0xc420356000, 0x0, 0x0, 0xc420174380, 0x32, 0x32)
May 24 17:03:18 hera journalbeat[2202]:         /tmp/nix-build-journalbeat-5.1.2.drv-0/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher/output.go:70 +0x9a
May 24 17:03:18 hera journalbeat[2202]: github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher.(*messageWorker).onEvent(0xc42015e960, 0x0, 0x1, 0xe25020, 0xc420356000, 0x0, 0x0, 0xc420174380, 0x32, 0x32)
May 24 17:03:18 hera journalbeat[2202]:         /tmp/nix-build-journalbeat-5.1.2.drv-0/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher/worker.go:83 +0x74
May 24 17:03:18 hera journalbeat[2202]: github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher.(*messageWorker).run(0xc42015e960)
May 24 17:03:18 hera journalbeat[2202]:         /tmp/nix-build-journalbeat-5.1.2.drv-0/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher/worker.go:69 +0x290
May 24 17:03:18 hera journalbeat[2202]: created by github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher.(*messageWorker).init
May 24 17:03:18 hera journalbeat[2202]:         /tmp/nix-build-journalbeat-5.1.2.drv-0/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher/worker.go:57 +0x103
May 24 17:03:18 hera systemd[1]: journalbeat.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
May 24 17:03:18 hera systemd[1]: journalbeat.service: Unit entered failed state.
May 24 17:03:18 hera systemd[1]: journalbeat.service: Failed with result 'exit-code'.
```

This looks very similar to: https://github.com/elastic/beats/issues/1378.

Hi!

Same issue here but indeed it's a bit strange, this issue happen only on some nodes of my cluster (with the same configuration on all nodes).

```
# ./journalbeat-debian -c /etc/journalbeat.yml -path.data /var/lib/journalbeat/ -path.home /var/lib/journalbeat/ -e
2017/07/04 14:46:31.484255 beat.go:285: INFO Home path: [/var/lib/journalbeat/] Config path: [/var/lib/journalbeat/] Data path: [/var/lib/journalbeat/] Logs path: [/var/lib/journalbeat//logs]
2017/07/04 14:46:31.484611 beat.go:186: INFO Setup Beat: journalbeat; Version: 5.4.1
2017/07/04 14:46:31.484419 metrics.go:23: INFO Metrics logging every 1m0s
2017/07/04 14:46:31.509758 add_cloud_metadata.go:231: INFO add_cloud_metadata: hosting provider type not detected.
2017/07/04 14:46:31.511935 client.go:123: INFO Elasticsearch url: http://schinf-c2.domain.net:9200
2017/07/04 14:46:31.512217 outputs.go:108: INFO Activated elasticsearch as output plugin.
2017/07/04 14:46:31.513258 publish.go:295: INFO Publisher name: journalbeat
2017/07/04 14:46:31.514000 async.go:63: INFO Flush Interval set to: 1s
2017/07/04 14:46:31.514213 async.go:64: INFO Max Bulk Size set to: 50
2017/07/04 14:46:31.518018 journalbeat.go:50: INFO Seek to tail successful
2017/07/04 14:46:31.521197 beat.go:221: INFO journalbeat start running.
2017/07/04 14:46:31.521245 journalbeat.go:129: INFO Journalbeat is running!
2017/07/04 14:46:31.547858 state.go:130: INFO Loaded 35 events, trying to publish
2017/07/04 14:46:32.247858 client.go:658: INFO Connected to Elasticsearch version 5.4.2
panic: interface conversion: interface {} is string, not common.Time

goroutine 52 [running]:
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch.getIndex(0xc420272300, 0x147ec80, 0xc42006b200, 0xc42018d620, 0xb)
	/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch/client.go:367 +0x57e
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch.createEventBulkMeta(0x147ec80, 0xc42006b200, 0xc42018d570, 0xc420272300, 0x0, 0x0, 0x0)
	/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch/client.go:339 +0x2bf
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch.bulkEncodePublishRequest(0x7f497b68a120, 0xc420078198, 0x147ec80, 0xc42006b200, 0xc42018d570, 0xc42006f180, 0x32, 0x32, 0xc8, 0x75d3c2, ...)
	/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch/client.go:289 +0xeb
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch.(*Client).PublishEvents(0xc4200de780, 0xc42006f180, 0x32, 0x32, 0xc3afc8, 0xc4200de780, 0x14f46b0400, 0x0, 0x0)
	/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch/client.go:241 +0x1a7
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single.(*Mode).PublishEvents.func1(0xc420193cb0)
	/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single/single.go:89 +0x100
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single.(*Mode).publish(0xc420193cb0, 0x1485440, 0xc4203710c0, 0x75d301, 0xc42009bb28, 0x14af008, 0x0)
	/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single/single.go:144 +0x402
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single.(*Mode).PublishEvents(0xc420193cb0, 0x1485440, 0xc4203710c0, 0xf40501, 0xc42006f180, 0x32, 0x32, 0xc420219f60, 0x1)
	/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single/single.go:99 +0xcd
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch.(*elasticsearchOutput).BulkPublish(0xc420074e00, 0x1485440, 0xc4203710c0, 0x1, 0xc42006f180, 0x32, 0x32, 0xc42003afb8, 0xc4201769c0)
	/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch/output.go:415 +0x9f
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher.(*outputWorker).sendBulk(0xc42007b310, 0xc42003ad38, 0xc42006f180, 0x32, 0x32)
	/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher/output.go:112 +0x1cb
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher.(*outputWorker).onBulk(0xc42007b310, 0xc42003ad38, 0xc42006f180, 0x32, 0x32)
	/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher/output.go:88 +0xbf
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher.(*outputWorker).onMessage(0xc42007b310, 0x0, 0x1, 0x1485440, 0xc4203710c0, 0x0, 0x0, 0xc42006f180, 0x32, 0x32)
	/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher/output.go:70 +0xad
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher.(*messageWorker).onEvent(0xc42007b310, 0x0, 0x1, 0x1485440, 0xc4203710c0, 0x0, 0x0, 0xc42006f180, 0x32, 0x32)
	/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher/worker.go:83 +0xa8
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher.(*messageWorker).run(0xc42007b310)
	/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher/worker.go:69 +0x2f7
created by github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher.(*messageWorker).init
	/go/src/github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/publisher/worker.go:57 +0x17d
```

With elasticsearch version 5.4.2 and this config for journalbeat:

```yaml
 ---
name: journalbeat
tags: [ "int2", "journalbeat" ]
logging.level: "info"
logging.to_files: false
logging.to_syslog: true
logging.metrics.enabled: true
logging.metrics.period: 60s
input:
  seek_position: cursor
  cursor_seek_fallback: tail
  write_cursor_state: true
  cursor_state_file: /var/lib/journalbeat/cursor.state
  flush_cursor_secs: 3
  clean_field_names: true
  convert_to_numbers: false
  move_metadata_to_field: journal
processors:
- add_cloud_metadata:
output:
  elasticsearch:
    enabled: true
    hosts: ['schinf-c2.domain.net:9200']
    index: "filebeat-%{+yyyy.MM.dd}"
    pipeline: "journalbeat"
    template.enabled: false
    template.versions.2x.enabled: false
```
I don't modify the `@timestamp` and I also didn't found why on certain nodes it's works well (same bin/systemd version…)

--- a/beater/journalbeat.go
+++ b/beater/journalbeat.go
@@ -136,6 +136,9 @@ func (jb *Journalbeat) publishPending() error {
 
 	logp.Info("Loaded %d events, trying to publish", len(pending))
 	for cursor, event := range pending {
+		// We need to convert the timestamp back to the correct type before trying to publish
+		timestamp, _ := time.Parse(time.RFC3339, event["@timestamp"].(string))
+		event["@timestamp"] = common.Time(timestamp)
 		ref := &eventReference{cursor, event}
 		jb.pending <- ref
 		refs = append(refs, ref)

